### PR TITLE
Fix cropping when rendering video

### PR DESCRIPTION
**Pull request by roomrys**
### Description
Although we allowed users to attempt to export a cropped version of their video - and had most of the code set-up to do so - there were still a few bugs making the crop feature unusable.

This PR allows users to render a cropped video that attempts to track instances in the frame - sometimes if instances move too fast, the crop does not follow in time due to smoothing or the crop may be too tiny to fit the instances.

### Types of changes

- [x] Bugfix
- [x] New feature (doubt this feature ever worked beforehand)
- [ ] Refactor / Code style update (no logical changes)
- [ ] Build / CI changes
- [ ] Documentation Update
- [ ] Other (explain)

### Does this address any currently open issues?
Render Video > Crop Size: Half results in cv2.error #672

### Outside contributors checklist

- [ ] Review the [guidelines for contributing](https://github.com/talmolab/sleap/wiki/Developer-Guide) to this repository
- [ ] Read and sign the [CLA](https://github.com/talmolab/sleap/blob/develop/sleap-cla.pdf) and add yourself to the [authors list](https://github.com/talmolab/sleap/blob/develop/AUTHORS)
- [ ] Make sure you are making a pull request against the **develop** branch (not *main*). Also you should start *your branch* off *develop*
- [ ] Add tests that prove your fix is effective or that your feature works
- [ ] Add necessary documentation (if appropriate)

#### Thank you for contributing to SLEAP!
:heart:

### CODE REVIEW

File: sleap/io/visuals.py [hunk 1]
```diff
@@ -264,21 +268,20 @@ def _mark_single_frame(self, video_frame: np.ndarray, frame_idx: int) -> np.ndar
...
         """
-
         # Use OpenCV to convert to BGR color image
         video_frame = img_to_cv(video_frame)
 
         # Add the instances to the image
         overlay = self._plot_instances_cv(video_frame.copy(), frame_idx)
 
-        return cv2.addWeighted(overlay, self.alpha, video_frame, 1 - self.alpha, 0)
+        return overlay
```
Comment (talmo) [inline sleap/io/visuals.py:277]: Why are we removing this?

This is necessary for overlaying the tracking with transparency on the original image. Without it, the markers and lines are at full opacity, so it's not as clear what's underneath them.

Comment (roomrys) [reply sleap/io/visuals.py]: Added this back to get some transparency

File: sleap/io/visuals.py [hunk 2]
```diff
@@ -112,34 +112,38 @@ def writer(
...
+            total_frames_written += len(data)
+            total_elapsed = perf_counter() - start_time
+            progress_queue.put((total_frames_written, total_elapsed))
+    except Exception as e:
+        raise (e)
+    finally:
+        if writer_object is not None:
+            writer_object.close()
     # send (-1, time) to signal done
     progress_queue.put((-1, total_elapsed))
```
Comment (roomrys) [inline sleap/io/visuals.py:148]: We are still able to get stuck in the code somewhere even with this `try except finally` executing as expected. For example, I intentionally errored-out to this message but was not returned to the CLI:
```
(sleap_render-video-crop) λ python tests\liezl_test_script.py
Saving config: C:\Users\TalmoLab/.sleap/1.2.4/preferences.yaml
Writing video with 14100 frame images...
INFO:sleap.io.visuals:Chunks: 221, chunk size: 64
Done in 0.41996690000000036 s, fps = 33574.07452825446.
Exception in thread Thread-3:
Traceback (most recent call last):
  File "C:\Users\TalmoLab\miniconda3\envs\sleap_render-video-crop\lib\threading.py", line 926, in _bootstrap_inner
    self.run()
  File "C:\Users\TalmoLab\miniconda3\envs\sleap_render-video-crop\lib\threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "d:\social-leap-estimates-animal-poses\pull-requests\sleap_render-video-crop\sleap\sleap\io\visuals.py", line 145, in writer
    raise (e)
  File "d:\social-leap-estimates-animal-poses\pull-requests\sleap_render-video-crop\sleap\sleap\io\visuals.py", line 126, in writer
    h, w = data[0].shape[:2]
AttributeError: 'NoneType' object has no attribute 'shape'
```

Comment (roomrys) [reply sleap/io/visuals.py]: Opting to push the crop video fixes in this PR and handle the threading hang-up as a different issue.

**Pull request merged**